### PR TITLE
feat: enable splitting of certificate secrets

### LINKED ISSUE #828: Split pfx files into cert and key
**Describe the solution you'd like**
We are migrating from acs-keyvault-agent to CSI driver. The [aforementioned solution](https://github.com/Hexadite/acs-keyvault-agent/blob/master/app/main.py#L346) splits pfx files into two parts: the cert and the key; this is useful for webservers which read the files separately. Users specify the secret name, cert filename, and key filename. Currently, the key and cert are concatenated.

**Anything else you would like to add:**
Concatenation occurs [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/35cc84ffb9f36f33d1c0b804c477d6d4ebdbe536/pkg/provider/provider.go#L560) and files are specified [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/35cc84ffb9f36f33d1c0b804c477d6d4ebdbe536/pkg/provider/provider.go#L290).

**Environment:**

- Secrets Store CSI Driver version: (use the image tag): latest
- Azure Key Vault provider version: (use the image tag): latest
- Kubernetes version: (use `kubectl version`): 1.21+
- Cluster type: (e.g. AKS, aks-engine, etc): AKS

### PR DESCRIPTION
**Pull request by rguptar**
**Reason for Change**:
This PR adds support for splitting a pfx/pkcs12 secret into its private key and cert chain. This will be useful for folks migrating from [keyvault-agent](https://github.com/Hexadite/acs-keyvault-agent) to CSI Driver as the former supports this.

**Is this a chart or deployment yaml update?**
No.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:
Fixes #828 

**Does this change contain code from or inspired by another project?**
- [ ] Yes
- [x] No

### CODE REVIEW

File: pkg/provider/provider.go [hunk 1]
```diff
@@ -91,6 +93,10 @@ type KeyVaultObject struct {
 	ObjectName string `json:"objectName" yaml:"objectName"`
 	// the filename the object will be written to
 	ObjectAlias string `json:"objectAlias" yaml:"objectAlias"`
+	// the key filename the secret object will be written to
+	ObjectKeyFile string `json:"objectKeyFile" yaml:"objectKeyFile"`
+	// the cert filename the secret object will be written to
+	ObjectCertFile string `json:"objectCertFile" yaml:"objectCertFile"`
```
Comment (nilekhc) [inline pkg/provider/provider.go:99]: Can you add these new fields in docs? 
https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/website/content/en/getting-started/usage/_index.md?plain=1#L64

File: pkg/provider/provider_test.go [hunk 2]
```diff
@@ -209,7 +209,7 @@ lKn75l/9h0PwiiPaI0TGKN2O8AwvhGGwDElmFhYtXedbbaST6rbVRDUj
 			if err != nil {
 				t.Fatalf("expected nil err, got: %v", err)
 			}
-			if content != tc.expectedKey+tc.expectedCert {
+			if content != tc.expectedKey+certKeySeparator+tc.expectedCert {
```
Comment (nilekhc) [inline pkg/provider/provider_test.go:212]: Could you also add a test which validates split on `certKeySeparator`

**Comment by nilekhc** [comment]
@rguptar Could you take a look at review comments? Let us know if we can help with something. Also the PR needs rebase.

File: pkg/provider/provider.go (hunk 1 above)
Comment (rguptar) [reply pkg/provider/provider.go:99]: updated

File: pkg/provider/provider_test.go (hunk 2 above)
Comment (rguptar) [reply pkg/provider/provider_test.go:212]: added unit test

**Comment by rguptar** [comment]
I've updated the PR but I'm a bit conflicted whether this PR should be merged. Modern servers should support using the combined certkey for both key and cert (e.g. NGINX). My codebase was doing manual post-processing of the key and cert (which I've since removed) and this should be highly discouraged. This PR may offer a peace of mind to people migrating from keyvault-agent but isn't strictly necessary.

**Comment by aramase** [comment]
> I've updated the PR but I'm a bit conflicted whether this PR should be merged. Modern servers should support using the combined certkey for both key and cert (e.g. NGINX). My codebase was doing manual post-processing of the key and cert (which I've since removed) and this should be highly discouraged. This PR may offer a peace of mind to people migrating from keyvault-agent but isn't strictly necessary.

Thanks @rguptar! I'm in split minds too if this needs to be included. Agreed that most servers should be able to parse the single file to extract the cert and key. I'm glad you have made the changes in your code base as well. 

I would prefer not including a one off change just for migration. Given the current behavior of the provider has worked well and there haven't been a lot of ask for this feature, I'm inclined to close this PR now and revisit if we get more of the similar requests.

@rguptar @nilekhc Thoughts?

**Review by aramase — COMMENTED** [review]
> I've updated the PR but I'm a bit conflicted whether this PR should be merged. Modern servers should support using the combined certkey for both key and cert (e.g. NGINX). My codebase was doing manual post-processing of the key and cert (which I've since removed) and this should be highly discouraged. This PR may offer a peace of mind to people migrating from keyvault-agent but isn't strictly necessary.

Thanks @rguptar! I'm in split minds too if this needs to be included. Agreed that most servers should be able to parse the single file to extract the cert and key. I'm glad you have made the changes in your code base as well. 

I would prefer not including a one off change just for migration. Given the current behavior of the provider has worked well and there haven't been a lot of ask for this feature, I'm inclined to close this PR now and revisit if we get more of the similar requests.

@rguptar @nilekhc Thoughts?

**Comment by nilekhc** [comment]
> > I've updated the PR but I'm a bit conflicted whether this PR should be merged. Modern servers should support using the combined certkey for both key and cert (e.g. NGINX). My codebase was doing manual post-processing of the key and cert (which I've since removed) and this should be highly discouraged. This PR may offer a peace of mind to people migrating from keyvault-agent but isn't strictly necessary.
> 
> Thanks @rguptar! I'm in split minds too if this needs to be included. Agreed that most servers should be able to parse the single file to extract the cert and key. I'm glad you have made the changes in your code base as well.
> 
> I would prefer not including a one off change just for migration. Given the current behavior of the provider has worked well and there haven't been a lot of ask for this feature, I'm inclined to close this PR now and revisit if we get more of the similar requests.
> 
> @rguptar @nilekhc Thoughts?

@aramase Aggred. Let's close this PR and look out for similar ask from more user.

**Comment by aramase** [comment]
@rguptar You good with us closing this PR?

**Comment by rguptar** [comment]
Yes, I agree that it makes sense to close it.

**Pull request closed without merging**